src/Pelagos/Bundle/AppBundle/Controller/UI/StatsController.php

```diff
@@ -217,11 +217,11 @@ public function getDatasetSizeRangesAction()
             $qb->join('d.datasetSubmission', 'ds');
 
             if (array_key_exists('range0', $range)) {
-                $qb->andWhere('ds.datasetFileSize > :range0');
+                $qb->andWhere('COALESCE(ds.datasetFileColdStorageArchiveSize, ds.datasetFileSize) > :range0');
```

Comment:
COALESCE returns the first argument that is not null. But both fields could be filed out, and a dataset is not in cold storage?

Reply:
As far as I understand the way it that it would be used, the cold storage attributes would only be set if and only if a dataset was in cold storage. Conversion back to normal storage would entail nulling out the values for the cold storage attributes. I think we'll have to discuss this with the PO and see if anything else is needed.

Reply:
This should work fine with PELAGOS-3471 as that feature will allow for the removal of the Cold Storage attributes. There should be no case where a Cold Storage attribute is to be non-null and "inactive."